plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,609 @@
+		acc.AddFields(measurementMasters, sentinelMastersFields, sentinelMastersTags)
+
+		gatherReplicaStats(client, acc, masterName)
+		gatherSentinelStats(client, acc, masterName)
+	}
+}
+
+func gatherReplicaStats(
+	client *RedisSentinelClient,
+	acc telegraf.Accumulator,
+	masterName string,
+) {
+	replicasCmd := redis.NewSliceCmd("sentinel", "replicas", masterName)
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(replicasCmd)
+
+	replicas, replicasErr := replicasCmd.Result()
+	if replicasErr != nil {
+		acc.AddError(replicasErr)
+		return
+	}
+
+	for _, replica := range replicas {
+		if replica, replicaOk := replica.([]interface{}); replicaOk {
+			rm := toMap(replica)
+
+			replicaTags, replicaFields := convertSentinelReplicaOutput(acc, client.BaseTags(), masterName, rm)
+			acc.AddFields(measurementReplicas, replicaFields, replicaTags)
+		}
+	}
+}
+
+func gatherSentinelStats(
+	client *RedisSentinelClient,
+	acc telegraf.Accumulator,
+	masterName string,
+) {
+	sentinelsCmd := redis.NewSliceCmd("sentinel", "sentinels", masterName)
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(sentinelsCmd)
+
+	sentinels, sentinelsErr := sentinelsCmd.Result()
+	if sentinelsErr != nil {
+		acc.AddError(sentinelsErr)
+		return
+	}
+
+	for _, sentinel := range sentinels {
+		if sentinel, sentinelOk := sentinel.([]interface{}); sentinelOk {
+			sm := toMap(sentinel)
+
+			sentinelTags, sentinelFields := convertSentinelSentinelsOutput(acc, client.BaseTags(), masterName, sm)
+			acc.AddFields(measurementSentinels, sentinelFields, sentinelTags)
+		}
+	}
+}
+
+// converts `sentinel masters <name>` output to tags and fields
+func convertSentinelMastersOutput(
+	acc telegraf.Accumulator,
+	globalTags map[string]string,
+	master map[string]string,
+	quorumErr error,
+) (map[string]string, map[string]interface{}) {
+	tags := make(map[string]string)
+	for k, v := range globalTags {
+		tags[k] = v
+	}
+
+	tags["master"] = master["name"]
+
+	fields := make(map[string]interface{})
+
+	fields["has_quorum"] = 0
+	if quorumErr == nil {
+		fields["has_quorum"] = 1
+	}
+
+	for key, val := range master {
+		key = strings.Replace(key, "-", "_", -1)
+
+		switch valType := measurementMastersFields[key]; valType {
+		case "float":
+			if val, err := strconv.ParseFloat(val, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "integer":
+			if val, err := strconv.ParseInt(val, 10, 64); err == nil {
+				fields[key] = val
+			} else {
+				acc.AddError(err)
+			}
+		case "string":
+			fields[key] = val
+		default:
+			continue
+		}
+	}
+
+	return tags, fields
+}
+
+// converts `sentinel sentinels <name>` output to tags and fields
+func convertSentinelSentinelsOutput(
+	acc telegraf.Accumulator,
+	globalTags map[string]string,
+	masterName string,
+	sentinelMaster map[string]string,
+) (map[string]string, map[string]interface{}) {
```

Comment:
Please make this function returning an error instead and add it to the accumulator in the caller instead of passing in `acc`.

Reply:
@srebhan Addressed in https://github.com/influxdata/telegraf/pull/10042/commits/4c8fd93b34c4f9f6983d8300c2aa3bbb3ed25846